arSam/migrations/202410181342-fixDataDiscrepancies.js

```diff
@@ -0,0 +1,603 @@
+const {
+  calculateVariance,
+  dynamoClient,
+  getOne,
+  incrementAndGetNextSubAreaID,
+  marshall,
+  runQuery,
+  TABLE_NAME,
+  TransactWriteItemsCommand,
+  getParks,
+  getSubAreas
+} = require('../layers/baseLayer/baseLayer.js');
+const { EXPORT_VARIANCE_CONFIG } = require('../layers/constantsLayer/constantsLayer.js');
+const { DateTime } = require('luxon');
+const _ = require('lodash');
```

Comment:
We don't need to use `_`, can use these instead:

```
diff.inOldPark.push(...oldParkArr.filter(item => !newParkArr.some(newItem => JSON.stringify(item) === JSON.stringify(newItem))));
```

```
function isEqual(obj1, obj2) {
  if (obj1 === obj2) return true;

  if (typeof obj1 !== 'object' || obj1 === null || typeof obj2 !== 'object' || obj2 === null) {
    return false;
  }

  const keys1 = Object.keys(obj1);
  const keys2 = Object.keys(obj2);

  if (keys1.length !== keys2.length) return false;

  for (let key of keys1) {
    if (!keys2.includes(key) || !isEqual(obj1[key], obj2[key])) {
      return false;
    }
  }

  return true;
}

if (!isEqual(oldOrcs, newOrcs)) {
  diffs = findDifferences(oldOrcs, newOrcs);
}
```

```
if (newData && Object.keys(newData).length > 0) {
  // newData is not empty
}
```

Reply:
Resolved in commit